compiler/rustc/src/main.rs

```diff
@@ -1,4 +1,6 @@
 #![feature(unix_sigpipe)]
+// We need this feature as it changes `dylib` linking behavior and allows us to link to `rustc_driver`.
+#![feature(rustc_private)]
```

Comment:
I'm surprised we didn't need it before to invoke `rustc_driver::main`

Reply:
We compile it with an unstable flag which both marks all unmarked items as rustc_private and allows calling rustc_private functions without enabling the feature gate.

Reply:
That's also the reason why we should be able to do this change without modifying the rustc/rustdoc/clippy/rustfmt binaries, right? (like also checking for `-Zforce-unstable-if-unmarked` when deciding to look up the statically linked libraries).

Reply:
That didn't work out as some tool binaries only indirectly use rustc libraries and thus didn't have `feature(rustc_private)`.